sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -108,6 +108,16 @@ object TestingUDT {
   }
 }
 
+object TestingValueClass {
+  case class IntWrapper(i: Int) extends AnyVal
+  case class StrWrapper(s: String) extends AnyVal
+
+  case class ValueClassData(
+    intField: Int,
+    wrappedInt: IntWrapper,
+    strField: String,
+    wrappedStr: StrWrapper)
```

Comment:
We might need a comment to describe what this class is look like in Java.
Seems like it has 2 int fields `intField`, `wrappedInt`, and 2 string fields `strField`, `wrappedStr`. I'm not sure it is the same in Scala 2.12, though.